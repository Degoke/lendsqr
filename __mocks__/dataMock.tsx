import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
