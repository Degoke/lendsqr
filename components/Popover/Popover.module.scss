.popover {
  position: absolute;
  z-index: 2;
}
