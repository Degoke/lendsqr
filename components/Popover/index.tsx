import { ReactNode } from 'react'
import Paper from '../Paper'
import styles from './Popover.module.scss'

type Props = {
  isOpen: boolean
  children: ReactNode
  className?: string
}

export default function Popover({ isOpen, className, children }: Props) {
  return (
    <div className={`${styles.popover} ${className}`}>
      {isOpen ? <Paper>{children}</Paper> : null}
    </div>
  )
}
