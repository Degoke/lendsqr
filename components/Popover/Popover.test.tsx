import { render, screen } from '@testing-library/react'
import Popover from './index'
import { mockCard, MockComponent, mockImage } from '../../__mocks__/dataMock'

describe('Popover tests', () => {
  it('renders Popover component unchanged', () => {
    const { container } = render(
      <Popover isOpen={false}>
        <MockComponent />
      </Popover>
    )
    expect(container).toMatchSnapshot()
  })
})
