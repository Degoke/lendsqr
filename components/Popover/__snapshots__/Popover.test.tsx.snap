// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover tests renders Popover component unchanged 1`] = `
<div>
  <div
    class="popover undefined"
  />
</div>
`;
